internal/privileged/container.go

```diff
@@ -0,0 +1,356 @@
+	}
+
+	// Setup directory sandbox volume will be mounted
+	sandboxPath := fmt.Sprintf(sandboxMountFormat, id)
+	if _, err := os.Stat(sandboxPath); err != nil {
+		if err := os.MkdirAll(sandboxPath, 0777); err != nil {
+			return nil, fmt.Errorf("failed to create mounted folder: %s", err)
+		}
+	}
+	path = sandboxPath
+
+	if err := mountSandboxVolume(ctx, path, s.Root.Path); err != nil {
+		return nil, fmt.Errorf("failed to bind payload directory on host: %s", err)
+	}
+
+	container.sandboxMount = path
+	go container.waitBackground()
+	return container, nil
+}
+
+// CreateProcess creates a process on the host, starts it, adds it to the containers
+// job object and then waits for exit.
+func (c *HostJobContainer) CreateProcess(ctx context.Context, config interface{}) (_ cow.Process, err error) {
+	conf, ok := config.(*hcsschema.ProcessParameters)
+	if !ok {
+		return nil, errors.New("unsupported process config passed in")
+	}
+
+	path, args := seperateArgs(conf.CommandLine)
+	absPath, err := findExecutable(path, c.sandboxMount)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find executable: %s", err)
+	}
+
+	token, err := processToken(conf.User)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create process token: %s", err)
+	}
+
+	cmd := &exec.Cmd{
+		Env:  os.Environ(),
+		Dir:  c.sandboxMount,
+		Path: absPath,
+		Args: args,
+		SysProcAttr: &syscall.SysProcAttr{
+			CreationFlags: windows.CREATE_NEW_PROCESS_GROUP,
+			Token:         syscall.Token(token),
+		},
+	}
+
+	process := newProcess(cmd)
+
+	var (
+		stdin  io.WriteCloser
+		stdout io.ReadCloser
+		stderr io.ReadCloser
+	)
+	// Create process pipes if asked for.
+	if conf.CreateStdInPipe {
+		stdin, err = process.cmd.StdinPipe()
+		if err != nil {
+			return nil, fmt.Errorf("failed to create stdin pipe: %s", err)
+		}
+		process.stdin = stdin
+	}
+
+	if conf.CreateStdOutPipe {
+		stdout, err = process.cmd.StdoutPipe()
+		if err != nil {
+			return nil, fmt.Errorf("failed to create stdout pipe: %s", err)
+		}
+		process.stdout = stdout
+	}
+
+	if conf.CreateStdErrPipe {
+		stderr, err = process.cmd.StderrPipe()
+		if err != nil {
+			return nil, fmt.Errorf("failed to create stderr pipe: %s", err)
+		}
+		process.stderr = stderr
+	}
+
+	defer func() {
+		if err != nil {
+			process.Close()
+			c.Release(ctx)
+		}
+	}()
+
+	// TODO (dcantah): Start this suspended, add to job, then unsuspend
+	if err = process.start(); err != nil {
+		return nil, fmt.Errorf("failed to start host process: %s", err)
+	}
+
+	if err = c.job.assign(process); err != nil {
+		return nil, fmt.Errorf("failed to assign process to job object: %s", err)
+	}
+
+	// Close initBlock, and only do this once for the init process. This signifies
+	// that we can start listening for the IOCP notification that all processes have
+	// exited in a job. This notification is what will close the container waitBlock
+	// if it hasn't been already by Close.
+	c.initDoOnce.Do(func() {
+		close(c.initBlock)
```

Comment:
This can be removed I think now that I'm looking at it. This was from a previous iteration